src/mono/mono/metadata/icall.c

```diff
@@ -8374,6 +8374,115 @@ ves_icall_System_Diagnostics_Debugger_Log (int level, MonoString *volatile* cate
 		mono_get_runtime_callbacks ()->debug_log (level, *category, *message);
 }
 
+#ifdef ENABLE_NETCORE
+#define EVENT_PIPE_DUMMY_PROVIDER_ID 1
+#define EVENT_PIPE_DUMMY_SESSION_ID 1
+#define EVENT_PIPE_DUMMY_EVENT_ID 1
+#define EVENT_PIPE_ERROR_SUCCESS 0
+#define EVENT_PIPE_INVALID_WAIT_HANDLE 0
+
+typedef enum _EventPipeSerializationFormat{
+	NetPerf,
+	NetTrace
+} EventPipeSerializationFormat;
+
+typedef struct _EventPipeProviderConfigurationNative {
+    gunichar2_ptr provider_name;
+    uint64_t keywords;
+    uint32_t logging_level;
+    gunichar2_ptr filter_data;
```

Comment:
nit: could just be `gunichar2 *` since this is normal code.   


The `foo_ptr` typedefs are mostly for the benefit of `HANDLES()` declarations in `icall-def-netcore.h` (because we need the method argument signature to be a list of single preprocessor tokens).